src/utilities/effects.ts

```diff
@@ -5393,14 +5396,55 @@ const effects = {
       const { confirm, value } = await showImportWorkspaceFileModal(
         workspace,
         workspaceContents,
+        inputLanguageName,
+        outputLanguageExtensions,
         startingPath,
         workspace,
         user,
       );
       if (confirm) {
-        const { files, targetDirectory } = value;
+        const { convertedFileExtension, filesToConvert, filesToUpload, shouldKeepOriginalFiles, targetDirectory } =
+          value as {
+            convertedFileExtension: string;
+            filesToConvert: File[];
+            filesToUpload: File[];
+            shouldKeepOriginalFiles: boolean;
+            targetDirectory: string;
+          };
+
+        let fileArray: File[] = [];
+
+        const convertedFiles: File[] = await Promise.all(
+          filesToConvert.map(async file => {
+            const matchingOutputLanguageExtension = outputLanguageExtensions.find(fileExtension =>
+              file.name.endsWith(`.${fileExtension.replace(/^\./, '')}`),
+            );
+
+            if (matchingOutputLanguageExtension) {
+              const fileName = file.name.replace(
+                matchingOutputLanguageExtension.replace(/^\./, ''),
+                convertedFileExtension.replace(/^\./, ''),
+              );
+              const lastModified = Date.now();
+              const content = await file.text();
+              const convertedContent = await toInputFormat(content);
+              const fileBlob = new Blob([convertedContent], { type: 'text/plain' });
+
+              return new File([fileBlob], fileName, { lastModified, type: 'text/plain' });
```

Comment:
```suggestion
              return new File([convertedContent], fileName, { lastModified, type: 'text/plain' });
```

Non-blocking, I _think_ it's sufficient to do this? but I'm not super familiar with this API.

Reply:
Good call! I've updated to just pass the string directly!